src/screens/SecretIconsScreen.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { FlatList, StyleSheet, View } from 'react-native';
+
+import { RegularText } from '../components';
+import { colors, Icon, normalize } from '../config';
+
+type TIconItem = {
+  iconName: string;
+};
+
+const iconNames = Object.keys(Icon);
+
+const IconItem = ({ iconName }: TIconItem) => {
+  const SelectedIcon = Icon[iconName];
+
+  return (
+    <View style={styles.iconField}>
+      <SelectedIcon size={normalize(35)} color={colors.secondary} />
+      <RegularText smallest>{iconName}</RegularText>
+    </View>
+  );
+};
+
+export const SecretIconsScreen = () => {
+  return (
+    <FlatList
+      data={iconNames}
+      renderItem={({ item }) => <IconItem iconName={item} />}
+      keyExtractor={(item) => item}
```

Comment:
what is the `item`, that is used as a key here? is this following best practices?

![image](https://github.com/smart-village-solutions/smart-village-app-app/assets/1942953/0c2886ab-bc51-493f-b050-169b30cd0545)

can we keep it or can you think of something else?

Reply:
adjusted it to keyExtractor like its common practice in our codebase [29186de](https://github.com/smart-village-solutions/smart-village-app-app/pull/914/commits/29186de714d420a15bf9284a4e17cbc553609203)

Reply:
I think I now got it right, before using the code you wrote above the the items in the flatlist hat no unique key.  `keyExtractor={(index) => index.toString()}` thats my new approach and this way all items should have a unique key since the index of each iconname in the array should be unique.

Reply:
And how come, that you are converting the index to a string?
Can you read https://reactnative.dev/docs/flatlist#keyextractor again please, what's the same as in my screenshot earlier.

> ...The default extractor checks item.key, then item.id, and then falls back to using the index, like React does.

This way i see no benefit in passing an index manually.

Another point is the problem with using index, which should be avoided. The following article points it out as well: https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66#:~:text=Always%20prefer%20using%20a%20unique,of%20sorting%2C%20filtering%20might%20happen.

Reply:
using `keyExtractor={(index) => index.id}` or `keyExtractor={(index) => index.key}` i get the warning below and the vs code error on both, therefore I chose index.toString() even tho it might not be the best solution
<img width="295" alt="image" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/105894697/3a8eb39a-e564-4832-bd87-f41120b6f1a8">
<img width="915" alt="image" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/105894697/6beef97c-8fe7-450d-a896-0817f355c23c">

Reply:
do you know what the passed param is? please log the variable you want to use. is it an object? does it have `id` or `key` prop? what do you think `index.toString()` results to?

Reply:
should be fixed now in https://github.com/smart-village-solutions/smart-village-app-app/pull/914/commits/74026a36793ec9e05d851202baaf819b672fe17c

Reply:
did you answer my questions for yourself? please do it in text here next time, so i can follow your thoughts and way for finding solutions. did you check our code for other `keyExtractor`s first? the latest code seems very familiar to me.

Reply:
in the process I forgot about the question, sorry for not answering it. I've checked your code before and in a call with Arda we've gone through the code together again to do the right changes since I had a mental block after quite some time.